python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -649,6 +650,17 @@ def cron_tick_passed(
         )
         return CronTickPassedCondition(cron_schedule=cron_schedule, cron_timezone=cron_timezone)
 
+    @public
+    @staticmethod
+    def freshness_state(status: FreshnessState) -> "BuiltinAutomationCondition":
```

Comment:
going back to the consistency with the check condition, the fact that `FreshnessState` isn't exported from top-level dagster makes this argument type a bit problematic. The way we get around that with the check condition is to just fork this out into `check_passed` and `check_failed` conditions. Should probably do something similar here.

Reply:
oh and yeah on the typing bit this would be `BuiltinAutomationCondition[AssetKey]` -- this is the most important one as it prevents people from doing stuff like:

```python

@dg.asset_check(automation_condition=dg.AutomationCondition.freshness_passed())

```

(at least from the type system perspective)